ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-items/workspaces-items.tsx

```diff
@@ -110,7 +110,7 @@ const WorkspacesItems = (props: Props) => {
             </Icon>
             <div>New Workspace</div>
           </ModifiedButton>
-          {props.workspaces.map(workspace => {
+          {props.workspaces.filter(workspace => workspace.id).map(workspace => {
```

Comment:
❓ Are we really wanting to not show workspaces without ids at all? Why not just staple a temporary id to the workspaces that don't have them to actually address the warning? Something like `props.workspaces.map(workspace => {id: generateId(), ...workspace})`

Reply:
I'm ok with the solution of giving them random values for `key` if they don't have an id.

If you check out the gif above you can see the new workspace has the author "guest" for a split second even though I'm logged in as admin. I figured we might as well not show the workspace in that split second.

Reply:
On second thought, the key thing is probably less risky. I'll go with that.